<?xml version="1.0"?>
<launch>

<!-- <include file="$(find explore_large_map)/launch/large_map.launch"/> -->

<!-- common param for astar_planner_pkg -->
<arg name="local_map_frame_name" value="base_link"/>
<arg name="global_map_frame_name" value="/odom"/>
<arg name="abso_global_map_frame_name" value="/abso_odom"/>
<arg name="allow_time_transmission_delay_ms_" value="1000"/>

<!-- mkz
<arg name="robot_length" value="4.93"/>
<arg name="robot_width" value="2.45"/>
<arg name="robot_wheelbase" value="2.85"/>
<arg name="base2back" value="1.100"/>
<arg name="minimum_turning_radius" value="5.5"/>
-->

<!-- toyataã€€-->
<arg name="robot_length" value="4.9"/>
<arg name="robot_width" value="1.95"/>
<arg name="robot_wheelbase" value="2.86"/>
<arg name="base2back" value="1.0"/>
<arg name="minimum_turning_radius" value="5"/>


<!-- private param for explore_large_map_pkg : /global_map,/explore_entry_map ; /move_base_simple/goal, /local_search_goal /-->
<arg name="re_planner_map_topic_name" value="/global_map"/>
<arg name="re_goal_pose_topic_name" value="/local_search_goal"/>

    <!-- for test astar actionlib
    <node pkg="astar_planner" type="astar_action_client_node" name="astar_action_client_node" output="screen" />
    -->

    <node pkg="astar_planner" type="astar_action_server_node" name="astar_action_server_node" output="screen" >
        <param name="global_map_frame_name" value="$(arg global_map_frame_name)" />
        <param name="local_map_frame_name" value="$(arg local_map_frame_name)" />
        <param name="send_path_topic_name" value="/global_path" />

        <param name="path_plan_update_cycle_" value="0.08" />
        <param name="distance_threhold_to_goal_" value="2" />

    </node>

    <node pkg="astar_planner" type="get_plan_server_node" name="get_plan_server_node" output="screen" >
            <param name="receive_planner_map_topic_name" value="$(arg re_planner_map_topic_name)" />
            <param name="global_map_frame_name" value="$(arg global_map_frame_name)" />
            <param name="local_map_frame_name" value="$(arg local_map_frame_name)" />
            <param name="abso_global_map_frame_name" value="$(arg abso_global_map_frame_name)" />


            <param name="allow_time_transmission_delay_ms_" value="$(arg allow_time_transmission_delay_ms_)" />
            <param name="path_frame" value="$(arg global_map_frame_name)" />
            <param name="allow_use_last_path" value="true" />
            <param name="use_wavefront_heuristic" value="true" />
            <param name="use_back" value="true" />
            <param name="allow_offset_distance" value="1.5" />
            <param name="angle_size" value="48" />
            <param name="minimum_turning_radius" value="$(arg minimum_turning_radius)" />
            <param name="obstacle_threshold" value="70" />
            <param name="goal_radius" value="1" />
            <param name="goal_angle" value="150" />
            <param name="robot_length" value="$(arg robot_length)" />
            <param name="robot_wheelbase" value="$(arg robot_wheelbase)" />
            <param name="robot_width" value="$(arg robot_width)" />
            <param name="base2back" value="$(arg base2back)" />
            <param name="curve_weight" value="1.05" />
            <param name="reverse_weight" value="3.0" />
    </node>

    <!--
    <node pkg="rviz" type="rviz" name="rviz_astar" args="-d $(find astar_planner)/rviz/astar_planner_service.rviz"/>
    -->
</launch>
